Formatter/TwigFilterTrait.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace whatwedo\CoreBundle\Formatter;
+
+use Twig\Markup;
+use Twig\TwigFilter;
+
+trait TwigFilterTrait
+{
+    protected function getFilterName(): string
+    {
+        $arr = explode('\\', self::class);
+        $filterName = array_pop($arr);
+
+        return $this->camel_to_snake($filterName);
+    }
+
+    public function getFilters()
+    {
+        return [
+            new TwigFilter($this->getFilterName(), [ $this, 'applyFilter']),
+        ];
+    }
+
+    public function applyFilter($value, array $options = []) {
+        $this->processOptions($options);
```

Comment:
These functions should be included in the Trait as `abstract`, so using the Trait enforces that they are defined in the target class.

Reply:
Yes. The applyFilter() does not make sense and should be "private". just refactored the filter callback function.

Reply:
I was referring to the calls to `$this->processOptions` and `$this->getHtml`, as these are not part of this trait. This will cause issues when incorrectly using the Trait. If we include abstract stubs for these methods, the IDE can show hints to the user regarding these missing functions.